### PR TITLE
fix epic! issue with thread unsafe hashset

### PR DESCRIPTION
**Pull request by Dyvinia**
fixes unavoidable and random crashes due to null strings in the hashset

2023-11-07_22-05-48_devenv

### CODE REVIEW

File: FrostyModSupport/FrostyModExecutor.cs [hunk 1]
```diff
@@ -272,7 +272,8 @@ private void ProcessModResources(IResourceContainer fmod)
                     resource.FillAssetEntry(bEntry);
 
                     addedBundles.TryAdd(bEntry.SuperBundleId, new HashSet<string>());
-                    addedBundles[bEntry.SuperBundleId].Add(bEntry.Name);
+                    lock (addedBundles[bEntry.SuperBundleId])
```
Comment (wannkunstbeikor) [inline FrostyModSupport/FrostyModExecutor.cs]: add braces and put the hashset in a variable please so the getter does not get called twice

**Comment by wannkunstbeikor** [comment]
also are you sure that fixed the issue and it was not sth else that was causing the null strings?

**Comment by Dyvinia** [comment]
people were having near consistent yet random frosty crashes when loading latest BF+ alpha and this fix solved it

**Comment by wannkunstbeikor** [comment]
> people were having near consistent yet random frosty crashes when loading latest BF+ alpha and this fix solved it

yeah thats why i asked, since it seemed to be random, it could still persist and they just havent got it yet

**Comment by Dyvinia** [comment]
havent had any crashes since, and it makes sense as editing the list/hashset in a non thread safe manner is unsafe rather than explicitly breaking something

**Pull request closed without merging**